chromium_src/components/captive_portal/content/captive_portal_service.cc

```diff
@@ -0,0 +1,12 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "components/captive_portal/content/captive_portal_service.h"
+
+#define DISABLED_FOR_TESTING \
+  DISABLED_FOR_TESTING;      \
+  true
+#include "src/components/captive_portal/content/captive_portal_service.cc"
+#undef DISABLED_FOR_TESTING
```

Comment:
is there any existing Chromium browsertest we can easily repurpose for our captive portal? Would be good to make sure it won't break if the `DISABLED_FOR_TESTING` flag check is removed upstream

Reply:
they all disabled when I tried to enable they crashed by another reasons

Reply:
I agree this needs some tests.

Reply:
Captive portal related browser tests are disabled due to crashes and unfortunately we do not run `components:components_unittests` which is directly related to this code. I copied ones that checks captive cases where the portal is enabled to a new file and added own to check we have url overriden. If we ever run on `components:components_unittests`(there many compilation errors now) then we could remove copied tests and use originals.